src/Uart/UartRegisters.h

```diff
@@ -71,41 +71,47 @@ typedef volatile struct UartRegisters
+               UART_TE,          // Transmitter FIFO empty: Indicates that the transmitter FIFO is empty.
+               UART_BR,          // Break received: Indicates that a BREAK has been received.
+               UART_OV,          // Overrun: Indicates that one or more character have been lost due to overrun.
+               UART_PE,          // Parity error: Indicates that a parity error was detected.
+               UART_FE,          // Framing error: Indicates that a framing error was detected.
+               UART_TH,          // Transmitter FIFO half-full: Indicates that the FIFO is less than half-full.
+               UART_RH,          // Receiver FIFO half-full: Indicates that at least half of the FIFO is holding data.
+               UART_TF,          // Transmitter FIFO full: Indicates that the Transmitter FIFO is full.
+               UART_RF,          // Receiver FIFO full: Indicates that the Receiver FIFO is full.
+               UART_TCNT,        // Transmitter FIFO count: Shows the number of data frames in the transmitter FIFO.
+               UART_RCNT         // Receiver FIFO count: Shows the number of data frames in the receiver FIFO.
+} apbuart_status_register_flags; // Status Register flags definition
+
+#define UART_CONTROL_OFFSET 0x08u
+typedef enum { UART_RE = 0,       // Receiver enable: If set, enables the receiver.
+               UART_CTRL_TE,      // Transmitter enable: If set, enables the transmitter.
+               UART_RI,           // Receiver interrupt enable: If set, interrupts are generated when a frame is received.
+               UART_TI,           // Transmitter interrupt enable: If set, interrupts are generated when a frame is transmitted.
+               UART_PS,           // Parity select: Selects parity polarity (0 = even parity, 1 = odd parity) (when implemented).
+               UART_CTRL_PE,      // Parity enable: If set, enables parity generation and checking (when implemented).
+               UART_LB = 7,       // Loop back: If set, loop back mode will be enabled.
+               UART_CRTL_TF = 9,  // Transmitter FIFO interrupt enable: When set, Transmitter FIFO level interrupts are enabled.
+               UART_CTRL_RF,      // Receiver FIFO interrupt enable: When set, Receiver FIFO level interrupts are enabled.
+               UART_DB,           // FIFO debug mode enable: When set, it is possible to read and write the FIFO debug register.
+               UART_FA = 31       // FIFOs available: Set to 1, read-only. Receiver and transmitter FIFOs are available.
+} apbuart_control_register_flags; // Control Register flags definition
+
+#define UART_CLKSCL_OFFSET  0x0Cu
+#define UART_CLKSCL_DIV     8u          // Clock divider value for baud rate generation
+#define UART_CLKSCL_SHIFT   4u          // Byte shift to convert 12-bit clkscl register value to 16-bit
+
+#define UART_FIFO_OFFSET    0x10u
+
+#define FLAG_SET 0x01u
```

Comment:
I don't think unprefixed macros should be in the header. Either prefix them with UART, or move to C.